flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutorGateway.java

```diff
@@ -195,4 +195,11 @@
 	 * @return Future which is completed with the {@link TransientBlobKey} of the uploaded file.
 	 */
 	CompletableFuture<TransientBlobKey> requestFileUpload(FileType fileType, @RpcTimeout Time timeout);
+
+	/**
+	 * Returns the fully qualified address of Metric Query Service on the TaskManager.
+	 *
+	 * @return Fully qualified (RPC) address of Metric Query Service on the TaskManager.
+	 */
+	String getMetricQueryServiceAddress();
```

Comment:
All methods of the `TaskExecutorGateway` represent an RPC. Thus, it should return a `CompletableFuture<String> foobar(@RpcTimeout Time timeout);`

Reply:
Good point.